File: src/main/java/com/eager/questioncloud/user/RefreshTokenRepositoryImpl.java
package com.eager.questioncloud.user;

import java.time.Duration;
import lombok.RequiredArgsConstructor;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
public class RefreshTokenRepositoryImpl implements RefreshTokenRepository {
    private final RedisTemplate<String, String> redisTemplate;
    public static final String REFRESH_TOKEN_KEY_PREFIX = "REFRESH-TOKEN-UID:";

    @Override
    public String get(Long uid) {
        return redisTemplate.opsForValue().get(REFRESH_TOKEN_KEY_PREFIX + uid);
    }

    @Override
    public void delete(String uid) {
        redisTemplate.delete(REFRESH_TOKEN_KEY_PREFIX + uid);
    }

    @Override
    public void save(String token, Long uid) {
        redisTemplate.opsForValue().set(REFRESH_TOKEN_KEY_PREFIX + uid, token, Duration.ofHours(24));
    }
}

File: qc-domain/qc-user/qc-user-core/src/main/kotlin/com/eager/questioncloud/user/infrastructure/repository/RefreshTokenRepositoryImpl.kt
package com.eager.questioncloud.user.infrastructure.repository

import org.springframework.data.redis.core.RedisTemplate
import org.springframework.stereotype.Repository
import java.time.Duration

@Repository
class RefreshTokenRepositoryImpl(
    private val redisTemplate: RedisTemplate<String, String>
) : RefreshTokenRepository {

    override fun getByUserId(uid: Long): String? {
        return redisTemplate.opsForValue().get(REFRESH_TOKEN_KEY_PREFIX + uid)
    }

    override fun delete(uid: String) {
        redisTemplate.delete(REFRESH_TOKEN_KEY_PREFIX + uid)
    }

    override fun save(token: String, uid: Long) {
        redisTemplate.opsForValue().set(REFRESH_TOKEN_KEY_PREFIX + uid, token, Duration.ofHours(24))
    }

    companion object {
        const val REFRESH_TOKEN_KEY_PREFIX: String = "REFRESH-TOKEN-UID:"
    }
}
